apps/astarte_realm_management/lib/device_remover.ex

```diff
@@ -0,0 +1,207 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 20230 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+defmodule Astarte.RealmManagement.DeviceRemover do
+  alias Astarte.RealmManagement.Queries
+
+  use Task
+  require Logger
+  alias Astarte.Core.InterfaceDescriptor
+
+  @spec start_link(list) :: {:ok, pid}
+  def start_link(args) do
+    Task.start_link(__MODULE__, :run, args)
+  end
+
+  def run(args) do
+    realm_name = Keyword.fetch(args, :realm_name)
+    device_id = Keyword.fetch(args, :device_id)
```

Comment:
`Keyword.fetch` returns an `:ok/:error` tuple, I guess that's not what you want here